packages/core/components/InputOrGroup/styles.module.css

```diff
@@ -33,109 +32,143 @@
 }
 
 .Input-labelIcon {
-  color: var(--puck-color-grey-6);
+  color: var(--puck-color-grey-07);
   margin-right: 4px;
   padding-left: 4px;
 }
 
 .Input-disabledIcon {
-  color: var(--puck-color-grey-4);
+  color: var(--puck-color-grey-05);
   margin-left: auto;
 }
 
 .Input-input {
-  background: white;
+  background: var(--puck-color-white);
   border-width: 1px;
   border-style: solid;
-  border-color: var(--puck-color-grey-8);
+  border-color: var(--puck-color-grey-09);
   border-radius: 4px;
   font-family: inherit;
   font-size: 14px;
   padding: 12px 15px;
+  transition: border-color 50ms ease-in;
   width: 100%;
 }
 
 select.Input-input {
-  /* Safari */
-  appearance: none;
+  appearance: none; /* Safari */
   background: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='100' height='100' fill='%23c3c3c3'><polygon points='0,0 100,0 50,50'/></svg>")
     no-repeat;
   background-size: 12px;
   background-position: calc(100% - 12px) calc(50% + 3px);
   background-repeat: no-repeat;
-  background-color: white;
+  background-color: var(--puck-color-white);
+  cursor: pointer;
 }
 
-.Input--readOnly > .Input-input,
-.Input--readOnly > select.Input-input {
-  background-color: var(--puck-color-grey-11);
-  border-color: var(--puck-color-grey-8);
-  color: var(--puck-color-grey-5);
-  opacity: 1;
+@media (hover: hover) and (pointer: fine) {
+  .Input:has( > input):hover .Input-input:not([readonly]),
+  .Input:has( > textarea):hover .Input-input:not([readonly]) {
+    border-color: var(--puck-color-grey-05);
+    transition: none;
+  }
+  .Input:has( > select):hover .Input-input:not([disabled]) {
+    background-color: var(--puck-color-azure-12);
+    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='100' height='100' fill='%235a5a5a'><polygon points='0,0 100,0 50,50'/></svg>");
+    border-color: var(--puck-color-grey-05);
+    transition: none;
+  }
 }
 
-.Input-input:hover {
-  border-color: var(--puck-color-neutral-3);
+.Input-input:focus {
+  border-color: var(--puck-color-grey-05);
+  outline: 2px solid var(--puck-color-azure-05);
+  outline-offset: 2px;
+  transition: none;
 }
 
-.Input-input:focus {
-  border-color: var(--puck-color-azure-4);
-  outline: var(--puck-color-azure-8) 4px solid;
-  outline-offset: 0;
+.Input--readOnly > .Input-input,
+.Input--readOnly > select.Input-input {
+  background-color: var(--puck-color-grey-10);
+  border-color: var(--puck-color-grey-09);
+  color: var(--puck-color-grey-04);
+  cursor: default;
+  opacity: 1;
+  outline: 0;
+  transition: none;
 }
 
 .Input-radioGroupItems {
   display: flex;
-  border: 1px solid var(--puck-color-grey-7);
+  border: 1px solid var(--puck-color-grey-09);
   border-radius: 4px;
   flex-wrap: wrap;
-  overflow: hidden;
 }
 
 .Input-radio {
```

Comment:
Does the radio input need to be added to the tab index?

Reply:
It's in the tab index already because it uses HTML `input`element, and also because I switched it from being `display: none` to visually hidden CSS. I was able to show a focus state on the parent with `:focus-within`. When tabbing to the group, the checked item is focused, and then an alternative item can be selected with arrow keys (all built-in browser functionality).